api/v1beta1/openstackbaremetalset.go

```diff
@@ -144,12 +151,60 @@ func VerifyBaremetalSetScaleUp(
 				continue
 			}
 
-			l.Info("Available BaremetalHost", "BMH", baremetalHost.ObjectMeta.Name)
+			l.Info("Available BaremetalHost (compute labels not yet processed)", "BMH", baremetalHost.ObjectMeta.Name)
 
-			selectedBaremetalHosts[hostName] = baremetalHost
-			selectedCount++
+			availableBaremetalHosts = append(availableBaremetalHosts, baremetalHost)
+		}
+
+		// We only want to continue to individual compute label matching if we actually have
+		// enough BMHs remaining given the filtering above
+		if len(availableBaremetalHosts) >= newBmhsNeededCount {
+
+			// Now try to fit all the requested new compute hosts into the set of available BMHs
+			// given bmhLabelSelectors on each compute (if any) and the labels on the BMHs
+			//
+			// The problem we are trying to solve can be demonstrated through an example...
+			// Imagine we have 3 available BMHs at this point with the following simplified labels:
+			//
```

Comment:
I think the assumption I had with the existing logic is that labels for individual node selection would be unique and not overlapping  (i.e not have more than one BMH matching the same labels) and we will always find the first label match like k8s.